packages/meditrak-server/src/routes/importSurveys/importSurveys.js

```diff
@@ -72,21 +80,44 @@ const updateOrCreateDataElementInGroup = async (models, dataElementCode, dataGro
     service_type: serviceType,
     config,
   });
-  const dataElement = await models.dataSource.updateOrCreate(
-    {
+
+  // If dataElement already exists, don't overwrite it
+  let dataElement = await models.dataSource.findOne({ code: dataElementCode });
+
+  if (dataElement === null) {
+    dataElement = await models.dataSource.create({
       type: models.dataSource.getTypes().DATA_ELEMENT,
       code: dataElementCode,
-    },
-    { service_type: serviceType },
-  );
-  dataElement.config = { ...dataElement.config, ...config };
-  dataElement.sanitizeConfig();
-  await dataElement.save();
+      service_type: serviceType,
+    });
+    dataElement.config = { ...dataElement.config, ...config };
+    dataElement.sanitizeConfig();
+    await dataElement.save();
 
-  await models.dataElementDataGroup.findOrCreate({
-    data_element_id: dataElement.id,
-    data_group_id: dataGroup.id,
-  });
+    await models.dataElementDataGroup.findOrCreate({
+      data_element_id: dataElement.id,
+      data_group_id: dataGroup.id,
+    });
+  } else {
+    /*
+     * Link the existing dataElement to a potentially new dataGroup
+     *
+     * We use separate find and create instead of findOrCreate because we may have an
+     * "invalid" state, where the dataElement and dataGroup have different services, and
+     * we want to allow this to continue as is.
```

Comment:
Even with your well placed comment I can't understand why we can't use `findOrCreate`. Could you explain a bit more? Is there validation that runs on `findOrCreate` that doesn't run on `create` or something?

Reply:
Seems like an old comment that does not apply anymore, updated